csharp/extractor/Semmle.Extraction/Entity.cs

```diff
@@ -141,11 +142,19 @@ public static class ICachedEntityFactoryExtensions
         public static Entity CreateEntity2<Type, Entity>(this ICachedEntityFactory<Type, Entity> factory, Context cx, Type init)
             where Entity : ICachedEntity => cx.CreateEntity2(factory, init);
 
-        public static void DefineLabel(this IEntity entity, TextWriter trapFile)
+        public static void DefineLabel(this IEntity entity, TextWriter trapFile, IExtractor extractor)
```

Comment:
The errors I have seen were all in `WriteSubId` (that you handle below): why do we need to handle errors in this method as well?

Reply:
This is all about defence in depth. In theory this code is never called, however we want to make this method robust to errors and handle errors correctly. We don't want break the trap file.